api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -220,6 +228,13 @@ public List<Appointment> getAllAppointmentsReminder(String hours) {
 
     }
 
+    @Transactional
```

Comment:
This should be read-only Transactional(readOnly=true)

Reply:
Agreed, this is existing behavior, but we should change this to readOnly=true.  fyi @angshu